diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 2e640322b..15e1f7130 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -1544,7 +1544,7 @@ bool BackhaulManager::handle_slave_backhaul_message(int fd, ieee1905_1::CmduMess
     case beerocks_message::ACTION_BACKHAUL_REGISTER_REQUEST: {
 
         m_agent_fd      = fd;
-        auto agent_name = std::move(std::string("agent"));
+        auto agent_name = std::string("agent");
         LOG(DEBUG) << "Assigning FD (" << fd << ") to " << agent_name;
         m_cmdu_server->set_client_name(fd, agent_name);
 
diff --git a/agent/src/beerocks/slave/platform_manager/platform_manager.cpp b/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
index 42c5db280..6ffd8f637 100644
--- a/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
+++ b/agent/src/beerocks/slave/platform_manager/platform_manager.cpp
@@ -414,7 +414,7 @@ bool PlatformManager::handle_cmdu(int fd, ieee1905_1::CmduMessageRx &cmdu_rx)
             return false;
         }
 
-        auto agent_name = std::move(std::string("agent socket"));
+        auto agent_name = std::string("agent socket");
         LOG(DEBUG) << "Assigning FD (" << fd << ") to " << agent_name;
         m_cmdu_server->set_client_name(fd, agent_name);
 
diff --git a/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h b/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
index f8174e4bd..004575d69 100644
--- a/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
+++ b/common/beerocks/bcl/include/bcl/network/netlink_event_listener.h
@@ -10,6 +10,7 @@
 #define BCL_NETWORK_NETLINK_EVENT_LISTENER_H_
 
 #include <linux/netlink.h>
+#include <cstdint>
 
 #include <functional>
 #include <unordered_map>
diff --git a/common/beerocks/bwl/include/bwl/key_value_parser.h b/common/beerocks/bwl/include/bwl/key_value_parser.h
index f5cd993f1..cc2a71e2f 100644
--- a/common/beerocks/bwl/include/bwl/key_value_parser.h
+++ b/common/beerocks/bwl/include/bwl/key_value_parser.h
@@ -12,6 +12,7 @@
 #include <list>
 #include <string>
 #include <unordered_map>
+#include <cstdint>
 
 namespace bwl {
 
diff --git a/common/beerocks/bwl/shared/nl80211_socket.h b/common/beerocks/bwl/shared/nl80211_socket.h
index 916b9420b..9f3a3c33a 100644
--- a/common/beerocks/bwl/shared/nl80211_socket.h
+++ b/common/beerocks/bwl/shared/nl80211_socket.h
@@ -55,6 +55,9 @@ public:
      *
      * @return True on success and false otherwise.
      */
+    //CCNS
+    using netlink_socket::send_receive_msg;
+    
     virtual bool send_receive_msg(int command, int flags,
                                   std::function<bool(struct nl_msg *msg)> msg_create,
                                   std::function<void(struct nl_msg *msg)> msg_handle);
diff --git a/common/beerocks/multi_vendor/CMakeLists.txt b/common/beerocks/multi_vendor/CMakeLists.txt
index fd0f344a1..25e48fca9 100644
--- a/common/beerocks/multi_vendor/CMakeLists.txt
+++ b/common/beerocks/multi_vendor/CMakeLists.txt
@@ -35,7 +35,12 @@ add_library(${PROJECT_NAME} ${multi_vendor_sources})
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${prplmesh_VERSION} SOVERSION ${prplmesh_VERSION_MAJOR})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINK_FLAGS "-Wl,-z,defs")
 
-target_link_libraries(${PROJECT_NAME} PRIVATE ${MULTI_VENDOR_LIBS} bcl bwl bpl btlvf tlvf nbapi mapfcommon)
+if (PRPLMESH_ENABLE_AMBIORIX_UBUS_BACKEND)
+    message(STATUS "Ambiorix backend enabled")
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${MULTI_VENDOR_LIBS} bcl bwl bpl btlvf tlvf nbapi mapfcommon)
+else()
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${MULTI_VENDOR_LIBS} bcl bwl bpl btlvf tlvf mapfcommon)
+endif()
 
 # Install
 # New Vendor has to add the path of header files like example.
diff --git a/framework/platform/bpl/CMakeLists.txt b/framework/platform/bpl/CMakeLists.txt
index 55d1c7083..f6a045b28 100644
--- a/framework/platform/bpl/CMakeLists.txt
+++ b/framework/platform/bpl/CMakeLists.txt
@@ -182,11 +182,14 @@ else()
 
 endif()
 
-find_package(amxb REQUIRED)
-find_package(amxc REQUIRED)
-find_package(amxd REQUIRED)
-find_package(amxp REQUIRED)
-list(APPEND BPL_LIBS amxb amxc amxd amxp)
+if (PRPLMESH_ENABLE_AMBIORIX_UBUS_BACKEND)
+    find_package(amxb REQUIRED)
+    find_package(amxc REQUIRED)
+    find_package(amxd REQUIRED)
+    find_package(amxp REQUIRED)
+    
+    list(APPEND BPL_LIBS amxb amxc amxd amxp)
+endif()
 
 # BPL pwhm DM via wbapi
 if (USE_PRPLMESH_WHM)
diff --git a/framework/platform/bpl/cfg/dummy/bpl_cfg_dummy.cpp b/framework/platform/bpl/cfg/dummy/bpl_cfg_dummy.cpp
index dbe4f4885..794ec03ed 100644
--- a/framework/platform/bpl/cfg/dummy/bpl_cfg_dummy.cpp
+++ b/framework/platform/bpl/cfg/dummy/bpl_cfg_dummy.cpp
@@ -6,6 +6,7 @@
  * See LICENSE file for more details.
  */
 
+#ifdef USE_PRPLMESH_WHM
 #include <amxc/amxc.h>
 #include <amxp/amxp.h>
 
@@ -20,6 +21,7 @@
 
 #include <amxo/amxo.h>
 #include <amxo/amxo_save.h>
+#endif
 
 #include <bcl/beerocks_version.h>
 #include <bpl/bpl_cfg.h>
@@ -28,6 +30,7 @@
 namespace beerocks {
 namespace bpl {
 
+#ifdef USE_PRPLMESH_WHM
 /*
  * This function is to fetch data from DeviceInfo. 
  * data model. This file is used for legacy platforms.
@@ -69,6 +72,7 @@ bool get_string_value_dm(std::string &attr, std::string &value)
     amxc_var_clean(&data);
     return true;
 }
+#endif
 
 /*
  * Function to get the Serial Number from data model
@@ -76,12 +80,17 @@ bool get_string_value_dm(std::string &attr, std::string &value)
  */
 bool get_serial_number(std::string &serial_number)
 {
+#ifdef USE_PRPLMESH_WHM
     std::string attr = "SerialNumber";
 
     if (!get_string_value_dm(attr, serial_number)) {
         serial_number.assign("prplmesh12345");
     }
     return true;
+#endif
+    LOG(ERROR) << "Assigning dummy to serial_number due to Ambiorix dependency";
+    serial_number = "01010101";
+    return true;
 }
 
 /*
@@ -90,6 +99,7 @@ bool get_serial_number(std::string &serial_number)
  */
 bool get_software_version(std::string &software_version)
 {
+#ifdef USE_PRPLMESH_WHM
     std::string attr = "SoftwareVersion";
 
     if (!get_string_value_dm(attr, software_version)) {
@@ -97,6 +107,10 @@ bool get_software_version(std::string &software_version)
         software_version.assign(version_string);
     }
     return true;
+#endif
+    LOG(ERROR) << "Assigning dummy to software version due to Ambiorix dependency";
+    software_version = "01010101";
+    return true;
 }
 
 bool get_max_prioritization_rules(uint32_t &max_prioritization_rules)
diff --git a/framework/platform/bpl/cfg/linux/bpl_cfg_ifaces.cpp.in b/framework/platform/bpl/cfg/linux/bpl_cfg_ifaces.cpp.in
index 8e41f5817..fc931e229 100644
--- a/framework/platform/bpl/cfg/linux/bpl_cfg_ifaces.cpp.in
+++ b/framework/platform/bpl/cfg/linux/bpl_cfg_ifaces.cpp.in
@@ -7,6 +7,7 @@
  */
 
  #include <string>
+ #include <cstdint>
 
 namespace beerocks {
 namespace bpl {
diff --git a/vendor_message/src/beerocks_vendor_message_main.cpp b/vendor_message/src/beerocks_vendor_message_main.cpp
index d9fb8ef1e..83b1bb7a5 100644
--- a/vendor_message/src/beerocks_vendor_message_main.cpp
+++ b/vendor_message/src/beerocks_vendor_message_main.cpp
@@ -16,7 +16,9 @@
 #include <mapf/common/utils.h>
 #include <sys/types.h>
 
+#ifdef USE_PRPLMESH_WHM
 #include "ambiorix_impl.h"
+#endif
 #include "vendor_message_slave.h"
 
 // Do not use this macro anywhere else in ire process
