diff --git a/common/beerocks/bwl/CMakeLists.txt b/common/beerocks/bwl/CMakeLists.txt
index 0503d5982..cbccebe9f 100644
--- a/common/beerocks/bwl/CMakeLists.txt
+++ b/common/beerocks/bwl/CMakeLists.txt
@@ -115,20 +115,24 @@ elseif(BWL_TYPE STREQUAL "DWPAL")
 
 elseif(BWL_TYPE STREQUAL "NL80211")
 
-    file(GLOB HOSTAPD_SEARCH_PATHS "${PLATFORM_BUILD_DIR}/hostapd*/hostapd-*")
-    find_path(HOSTAPD_INCLUDE_DIR NAMES "src/common/wpa_ctrl.h" PATHS ${HOSTAPD_SEARCH_PATHS} NO_CMAKE_FIND_ROOT_PATH)
-    set(HOSTAPD_DIR "${HOSTAPD_INCLUDE_DIR}")
+message(STATUS "===================================")
+message(STATUS "PLATFORM_STAGING_DIR = ${PLATFORM_STAGING_DIR}")
+message(STATUS "===================================")
 
     find_package(nl-genl-3 REQUIRED)
     list(APPEND BWL_LIBS nl-genl-3)
 
+if(NOT EXISTS ${PLATFORM_STAGING_DIR})
+    message(FATAL_ERROR "Staging directory not found: ${PLATFORM_STAGING_DIR}")
+endif()
+
     file(GLOB bwl_platform_sources
         ${MODULE_PATH}/shared/*.c*
         ${MODULE_PATH}/nl80211/*.c*
 
         # WPA Control Interface
-        ${HOSTAPD_DIR}/src/common/wpa_ctrl.c
-        ${HOSTAPD_DIR}/src/utils/os_unix.c
+        ${PLATFORM_STAGING_DIR}/usr/include/wpa_ctrl.c
+        ${PLATFORM_STAGING_DIR}/usr/include/os_unix.c
     )
 
     # Enable WPA control interface
@@ -139,9 +143,7 @@ elseif(BWL_TYPE STREQUAL "NL80211")
 
     # Hostapd/NL80211 include directories
     include_directories(
-        ${HOSTAPD_DIR}/src/utils
-        ${HOSTAPD_DIR}/src/common
-        ${HOSTAPD_DIR}/src/drivers
+        ${PLATFORM_STAGING_DIR}/usr/include
     )
 
     # Platform libraries
