From fd01698a726e5448f4bba68e507bcf8784bd6bc3 Mon Sep 17 00:00:00 2001
From: Suvidh Subodh Hegde <suvidhsu@qti.qualcomm.com>
Date: Mon, 8 Sep 2025 18:26:35 +0530
Subject: [PATCH] Debug symbol enable size optimisation

Signed-off-by: Suvidh Subodh Hegde <suvidhsu@qti.qualcomm.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 109f4be38..2dada9e81 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,7 +118,7 @@ endif()
 
 # Default Compiler flags
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -pthread -Wno-comment -Wno-unused-parameter -Wno-missing-field-initializers -fno-exceptions")
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Wformat -Wformat-security -fPIC -fPIE -fno-exceptions")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O2 -Wformat -Wformat-security -fPIC -fPIE -fno-exceptions")
 
 # Compiler specific flags
 if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.1)
@@ -131,7 +131,7 @@ endif()
 # Release flags
 set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_CXX_FLAGS}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_C_FLAGS_RELEASE}")
-set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_RELEASE} -g3 -O0 -DDEBUG")
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_RELEASE} -g3")
 
 # Linker Options
 # - Setting the "rpath-link" linker variable to help the linker resolve secondery dependecies
-- 
2.34.1

