From 146940a4ca4744d41cf82108bec43514cb674f5d Mon Sep 17 00:00:00 2001
From: Vinu Bharath R <vinubhar@qti.qualcomm.com>
Date: Fri, 11 Apr 2025 17:43:40 +0530
Subject: [PATCH 4/7] Controller: APMLD ODL update

Signed-off-by: Vinu Bharath R <vinubhar@qti.qualcomm.com>
---
 controller/nbapi/odl/device.odl.in | 120 +++++++++++++++++++++++++++++
 1 file changed, 120 insertions(+)

diff --git a/controller/nbapi/odl/device.odl.in b/controller/nbapi/odl/device.odl.in
index 08e6bb419..1dc6052ef 100644
--- a/controller/nbapi/odl/device.odl.in
+++ b/controller/nbapi/odl/device.odl.in
@@ -308,5 +308,125 @@
         %read-only bool SupportsVBSS = false;
 
         %instance bool UpdateVBSSCapabilities();
+
+        /*
+        ** The list of APMLD objects.
+        */
+        object APMLD [] {
+            counted with APMLDNumberOfEntries;
+            %read-only string MLDMACAddress = "";
+
+            object APMLDConfig {
+                %read-only bool EMLMREnabled = false;
+                %read-only bool EMLSREnabled = false;
+                %read-only bool STREnabled = false;
+                %read-only bool NSTREnabled = false;
+                %read-only bool TIDLinkMapNegotiation = false;
+            }
+
+            object TIDLinkMap [] {
+                counted with TIDLinkMapNumberOfEntries;
+                %read-only string Direction {
+                    default "Down";
+                    on action validate call check_enum [
+                        "Up",
+                        "Down"
+                    ];
+                }
+                %read-only uint8 TID;
+                %read-only string BSSID = "";
+                %read-only uint8 LinkID;
+            }
+
+            /* Affiliated APs list of APMLD */
+            object AffiliatedAP [] {
+                counted with AffiliatedAPNumberOfEntries;
+                %read-only string BSSID = "";
+                %read-only uint8 LinkID;
+                %read-only string RUID = "";
+                %read-only uint16 DisabledSubChannels;
+                %read-only uint32 PacketsSent = 0;
+                %read-only uint32 PacketsReceived = 0;
+                %read-only uint32 UnicastBytesSent = 0;
+                %read-only uint32 UnicastBytesReceived = 0;
+                %read-only uint32 ErrorsSent = 0;
+                %read-only uint32 MulticastBytesSent = 0;
+                %read-only uint32 MulticastBytesReceived = 0;
+                %read-only uint32 BroadcastBytesSent = 0;
+                %read-only uint32 BroadcastBytesReceived = 0;
+                %read-only string EstServiceParametersBE;
+                %read-only string EstServiceParametersBK;
+                %read-only string EstServiceParametersVI;
+                %read-only string EstServiceParametersVO;
+            }
+
+            /* Associated STAMLD list (Client/bSTA) */
+            object STAMLD [] {
+                counted with STAMLDNumberOfEntries;
+                %read-only string MLDMACAddress = "";
+                %read-only string Hostname;
+                %read-only string IPv4Address;
+                %read-only string IPv6Address;
+                %read-only bool IsbSTA = false;
+                %read-only uint32 LastConnectTime = 0;
+                %read-only uint32 BytesSent = 0;
+                %read-only uint32 BytesReceived = 0;
+                %read-only uint32 PacketsSent = 0;
+                %read-only uint32 PacketsReceived = 0;
+                %read-only uint32 ErrorsSent = 0;
+                %read-only uint32 ErrorsReceived = 0;
+                %read-only uint32 RetransCount = 0;
+
+                object WiFi7Capabilities {
+                    %read-only bool EMLMRSupport = false;
+                    %read-only bool EMLSRSupport = false;
+                    %read-only bool STRSupport = false;
+                    %read-only bool NSTRSupport = false;
+                    %read-only bool TIDLinkMapNegotiation = false;
+                }
+
+                object STAMLDConfig {
+                    %read-only bool EMLMREnabled = false;
+                    %read-only bool EMLSREnabled = false;
+                    %read-only bool STREnabled = false;
+                    %read-only bool NSTREnabled = false;
+                    %read-only bool TIDLinkMapNegotiation = false;
+                }
+
+                object STATIDLinkMap [] {
+                    counted with STATIDLinkMapNumberOfEntries;
+                    %read-only string Direction {
+                        default "Down";
+                        on action validate call check_enum [
+                            "Up",
+                            "Down"
+                        ];
+                    }
+
+                    %read-only uint8 TID;
+                    %read-only string BSSID = "";
+                    %read-only uint8 LinkID;
+                }
+
+                /* Affiliated STAs list of STAMLD(Client/bSTA) */
+                object AffiliatedSTA [] {
+                    counted with AffiliatedSTANumberOfEntries;
+                    %read-only string MACAddress = "";
+                    %read-only string BSSID = "";
+                    %read-only uint32 BytesSent = 0;
+                    %read-only uint32 BytesReceived = 0;
+                    %read-only uint32 PacketsSent = 0;
+                    %read-only uint32 PacketsReceived = 0;
+                    %read-only uint32 ErrorsSent = 0;
+                    %read-only uint8 SignalStrength = 0;
+                    %read-only uint32 EstMACDataRateDownlink = 0;
+                    %read-only uint32 EstMACDataRateUplink = 0;
+                    %read-only uint32 LastDataDownlinkRate = 0;
+                    %read-only uint32 LastDataUplinkRate = 0;
+                    %read-only uint64 UtilizationReceive = 0;
+                    %read-only uint64 UtilizationTransmit = 0;
+                }
+            }
+        }
     }
 }
-- 
2.34.1

