From 235a293f592f10e7c4c3c4eceb79caee425d7ea4 Mon Sep 17 00:00:00 2001
From: Girija P <gpandiya@qti.qualcomm.com>
Date: Thu, 8 Jan 2026 00:30:30 +0530
Subject: [PATCH 30/54] Corrected patching errors on top of Jan 5 baseline

Signed-off-by: Girija P <gpandiya@qti.qualcomm.com>
---
 .../ap_manager/ap_manager.cpp                  | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp b/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp
index a4dd9fa6c..135adcc5c 100644
--- a/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp
+++ b/agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager.cpp
@@ -1978,15 +1978,6 @@ void ApManager::handle_cmdu(ieee1905_1::CmduMessageRx &cmdu_rx)
                 if (!all_vaps_enabled && ap_wlan_hal->get_vap_status(bss_info_conf_list)) {
                     LOG(INFO) << "All vaps are enabled";
                     all_vaps_enabled = true;
-                }
-
-                if (!all_mld_updated && ap_wlan_hal->update_mld_status(bss_info_conf_list)) {
-                    LOG(INFO) << "MLD fields are updated";
-                    all_mld_updated = true;
-                }
-
-                if (all_vaps_enabled && all_mld_updated)
-                    break;
                 } else {
 #ifdef USE_PRPLMESH_NLHAL
                     m_ap_hal_ext_events  = ap_wlan_hal->get_ext_events_fds();
@@ -1996,6 +1987,15 @@ void ApManager::handle_cmdu(ieee1905_1::CmduMessageRx &cmdu_rx)
                           }
                     }
 #endif
+                }
+
+                if (!all_mld_updated && ap_wlan_hal->update_mld_status(bss_info_conf_list)) {
+                    LOG(INFO) << "MLD fields are updated";
+                    all_mld_updated = true;
+                }
+
+                if (all_vaps_enabled && all_mld_updated)
+                    break;
                 UTILS_SLEEP_MSEC(500);
             }
             if (all_vaps_enabled == false || all_mld_updated == false) {
-- 
2.34.1

