From 4554b0f351a194cc6956c08b27d7121ccedb5e55 Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Wed, 22 Oct 2025 18:55:37 +0530
Subject: [PATCH] Fix for Prplmesh process after reboot

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>

diff --git a/common/beerocks/scripts/prplmesh_utils.sh.in b/common/beerocks/scripts/prplmesh_utils.sh.in
index a64f74d..d894fa8 100755
--- a/common/beerocks/scripts/prplmesh_utils.sh.in
+++ b/common/beerocks/scripts/prplmesh_utils.sh.in
@@ -222,6 +222,7 @@ roll_logs_function()
 }
 
 start_function() {
+    sleep 30
     echo "$0: start"
     [ "$(id -u)" -ne 0 ] && echo "$0: warning - this commands needs root privileges. It might not work without them (are you root?)"
 
-- 
2.34.1

