From aa4b91079e478a85224b8326fde44efd26c1b376 Mon Sep 17 00:00:00 2001
From: nushanth <nushanth@qti.qualcomm.com>
Date: Wed, 11 Feb 2026 17:57:46 +0530
Subject: [PATCH] QC Add AKM Suite Capabilities TLV

Signed-off-by: nushanth <nushanth@qti.qualcomm.com>
---
 .../slave/tasks/capability_reporting_task.cpp | 64 +++++++++++++++++++
 1 file changed, 64 insertions(+)

diff --git a/agent/src/beerocks/slave/tasks/capability_reporting_task.cpp b/agent/src/beerocks/slave/tasks/capability_reporting_task.cpp
index a1cb52e..a4f8e88 100644
--- a/agent/src/beerocks/slave/tasks/capability_reporting_task.cpp
+++ b/agent/src/beerocks/slave/tasks/capability_reporting_task.cpp
@@ -208,6 +208,70 @@ bool CapabilityReportingTask::add_akm_suites_capabilities_tlv(ieee1905_1::CmduMe
 
     // TODO: Fill correct values for TLV (PPM-3618)
 
+    auto db = AgentDB::get();
+
+    int radio_count = 0;
+    for (auto radio : db->get_radios_list()) {
+        if (radio)
+            radio_count++;
+    }
+
+    if (!akm_suites_capabilities_tlv->alloc_backhaul_bss_akm_suite_selectors(radio_count)) {
+        LOG(ERROR) << "Error unable to allocate " << radio_count << " backhaul suite selectors";
+    }
+
+    if (!akm_suites_capabilities_tlv->alloc_fronthaul_bss_akm_suite_selectors(radio_count)) {
+        LOG(ERROR) << "Error unable to allocate " << radio_count << " fronthaul suite selectors";
+    }
+
+    for (int idx = 0; idx < radio_count; idx++) {
+        auto bh_selector = akm_suites_capabilities_tlv->backhaul_bss_akm_suite_selectors(idx);
+        auto fh_selector = akm_suites_capabilities_tlv->fronthaul_bss_akm_suite_selectors(idx);
+
+        bool bh_ret = std::get<0>(bh_selector);
+        if (!bh_ret) {
+            LOG(ERROR) << "Error failed to get backhaul bss akm suite for idx " << idx;
+        }
+
+        bool fh_ret = std::get<0>(fh_selector);
+        if (!fh_ret) {
+            LOG(ERROR) << "Error failed to get fronthaul bss akm suite for idx " << idx;
+        }
+        if (bh_ret && fh_ret) {
+            auto &bh_suite_selector = std::get<1>(bh_selector);
+            auto &fh_suite_selector = std::get<1>(fh_selector);
+
+            switch (idx) {
+                case 0:
+		    bh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::IEEE80211);
+		    bh_suite_selector.oui.struct_swap();
+		    bh_suite_selector.akm_suite_type = 0x02;
+		    fh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::IEEE80211);
+		    fh_suite_selector.oui.struct_swap();
+		    fh_suite_selector.akm_suite_type = 0x02;
+		    break;
+
+                case 1:
+                    bh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::IEEE80211);
+		    bh_suite_selector.oui.struct_swap();
+		    bh_suite_selector.akm_suite_type = 0x08;
+		    fh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::IEEE80211);
+		    fh_suite_selector.oui.struct_swap();
+		    fh_suite_selector.akm_suite_type = 0x08;
+		    break;
+
+                case 2:
+                    bh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::WEC);
+		    bh_suite_selector.oui.struct_swap();
+		    bh_suite_selector.akm_suite_type = 0x02;
+		    fh_suite_selector.oui = sVendorOUI(wfa_map::tlvAkmSuiteCapabilities::eAkmSuiteOUI::WEC);
+		    fh_suite_selector.oui.struct_swap();
+		    fh_suite_selector.akm_suite_type = 0x02;
+		    break;
+	    }
+        }
+    }
+
     LOG(INFO) << "Add AKM Suites capabilities TLV";
     return true;
 }
-- 
2.34.1

